ko/news/_posts/2021-12-25-ruby-3-1-0-released.md

```diff
@@ -0,0 +1,257 @@
+
+최근에 Shopify는 그들의 Rails 애플리케이션의 속도를 높이기 위해 Ruby에 많은 기여를 했습니다. YJIT는 중요한 기여로, Rails 애플리케이션의 성능 향상을 목표로 합니다.
+
+MJIT가 메소드 기반의 JIT 컴파일러고 외부 C 컴파일러를 이용할 때, YJIT는 기본 블록 버전 관리를 이용하며 JIT 컴파일러를 포함합니다. 게으른 기본 블록 버전 관리(Lazy Basic Block Versioning, LBBV)를 통해 우선 메소드의 시작을 컴파일하고 인자와 변수의 타입이 동적으로 정의되면 점진적으로 나머지 부분을 컴파일합니다. 자세한 설명은 [YJIT: CRuby를 위한 기본 블록 버전 관리 JIT 컴파일러](https://dl.acm.org/doi/10.1145/3486606.3486781)를 참고하세요.
+
+이 기술을 이용하여, YJIT는 대부분의 실제 세계 소프트웨어들에 railsbench에서는 최대 22%, liquid-render에서는 최대 39%의 빠른 준비 시간과 성능 향상을 가져다줍니다.
+
+YJIT는 아직 실험적인 기능이기 때문에 기본으로는 비활성화되어있습니다. 만약 사용하고 싶다면 `--yjit` 명령어 옵션을 통해 활성화할 수 있습니다. 그리고 아직은 Unix-like한 x86-64 플랫폼으로만 한정되어있습니다.
+
+* <https://bugs.ruby-lang.org/issues/18229>
+* <https://shopify.engineering/yjit-just-in-time-compiler-cruby>
+* <https://www.youtube.com/watch?v=PBVLf3yfMs8>
+
+## debug gem: 새로운 디버거
+
+완전히 새로 쓰인 디버거인 [debug.gem](https://github.com/ruby/debug)이 포함됩니다. debug.gem은 다음 기능을 갖고 있습니다:
+
+* 디버깅 성능 향상 (디버거가 있어도 애플리케이션을 느리게 만들지 않음).
+* 원격 디버깅 지원.
+* 풍부한 프론트엔드 디버깅 지원 (이제 VSCode와 크롬 브라우저를 지원).
+* 멀티 프로세스/멀티 스레드 디버깅 지원.
+* REPL 색상 지원.
+* 그리고 녹화 & 재생 기능, 추적 기능 등 다양하고 유용한 기능들.
+
+<video autoplay="autoplay" controls="controls" muted="muted" width="764" height="510" poster="https://cache.ruby-lang.org/pub/media/ruby310_debug_demo.jpg">
+  <source src="https://cache.ruby-lang.org/pub/media/ruby310_debug_demo.mp4" type="video/mp4">
+</video>
+
+Ruby는 lib/debug.rb를 포함했지만 잘 관리되지 않고 성능과 기능에 여러 이슈가 있었습니다. debug.gem은 lib/debug.rb를 완전히 대체했습니다.
+
+## error_highlight: 백트레이스에서의 세밀한 에러 위치
+
+빌트인된 gem인 error_highlight가 소개되었습니다. 백트레이스에서 세밀한 에러 위치를 포함합니다.
+
+```
+$ ruby test.rb
+test.rb:1:in `<main>': undefined method `time' for 1:Integer (NoMethodError)
+
+1.time {}
+ ^^^^^
+Did you mean?  times
+```
+
+현재는 `NameError`만 지원합니다.
+
+이 gem은 기본적으로 활성화되어있습니다. 명령어에서 `--disable-error_highlight` 옵션을 통해 비활성화할 수 있습니다. [error_highlight 리포지토리](https://github.com/ruby/error_highlight)를 통해 자세한 내용을 참고하세요.
+
+## IRB 자동완성과 문서 표시
+
+IRB는 이제 그냥 코드를 입력하기만 하면 완성 후보 다이얼로그가 나타나는 자동 완성 기능을 포함합니다. 탭과 시프트+탭을 통해 위아래로 이동할 수 있습니다.
+
+만약에 완성 후보를 선택할 때 문서가 설치되어있다면 문서 다이얼로그가 선택한 완성 후보 다이얼로그 옆에 나타나며 해당 내용을 보여줍니다. 알트+d를 눌러서 전체 문서를 읽을 수 있습니다.
+
+<video autoplay="autoplay" controls="controls" muted="muted" width="764" height="510" poster="https://cache.ruby-lang.org/pub/media/ruby310_irb_dialog.png">
+  <source src="https://cache.ruby-lang.org/pub/media/ruby310_irb_dialog.mp4" type="video/mp4">
+</video>
+
+## 다른 주목할만한 새로운 기능들
+
+### 언어
+
+* 해시 리터럴의 값과 키워드 인자를 생략할 수 있습니다. [[Feature #14579]](https://bugs.ruby-lang.org/issues/14579)
+  * `{x:, y:}`는 `{x: x, y: y}`의 편의 문법입니다.
+  * `foo(x:, y:)`는 `foo(x: x, y: y)`의 편의 문법입니다.
+
+* 패턴 매칭에서 핀 연산자 (`^`)에 표현식을 사용할 수 있습니다. [[Feature #17411]](https://bugs.ruby-lang.org/issues/17411)
```

Comment:
```suggestion
* 패턴 매칭에서 핀 연산자(`^`)에 표현식을 사용할 수 있습니다. [[Feature #17411]](https://bugs.ruby-lang.org/issues/17411)
```

Reply:
There's a space in the English version. Is the English spelling rule different from than Korean rule?

https://github.com/ruby/www.ruby-lang.org/blob/312e842b3ce13fcbd9067f7180ebd75791ca3a38/en/news/_posts/2021-12-25-ruby-3-1-0-released.md?plain=1#L85

Reply:
Actually, yes. Please take a look at "10. 소괄호( (    ) )" section. I think this is case (1). https://kornorms.korean.go.kr/regltn/regltnView.do?regltn_code=0001&regltn_no=714

Also, when we read except parentheses, it will be "핀 연산자 에" instead of "핀 연산자에", but postpositional particles should not have space before them in Korean.

Reply:
Thanks! Do you use any Korean spell checkers?

Reply:
I'm not using spell checkers. For correct spacing, I search "○○ 띄어쓰기" to check [온라인가나다](https://www.korean.go.kr/front/onlineQna/onlineQnaList.do?mn_id=216). For correct spelling, I search [네이버 사전](https://dict.naver.com) to check 표준국어대사전. For punctuations, the above link was helpful. Also for some IT-related words, I do googling to check the frequency of use, and the spelling used by Korean Wikipedia.